uberenv.py

```diff
@@ -386,82 +389,44 @@ def config_dir(self):
         return spack_config_dir
 
 
-    def disable_spack_config_scopes(self,spack_dir):
-        # disables all config scopes except "defaults", which we will
-        # force our settings into
-        spack_lib_config = pjoin(spack_dir,"lib","spack","spack","config.py")
-        print("[disabling config scope (except defaults) in: {}]".format(spack_lib_config))
-        cfg_script = open(spack_lib_config).read()
-        for cfg_scope_stmt in ["('system', os.path.join(spack.paths.system_etc_path, 'spack')),",
-                            "('site', os.path.join(spack.paths.etc_path, 'spack')),",
-                            "('user', spack.paths.user_config_path)"]:
-            cfg_script = cfg_script.replace(cfg_scope_stmt,
-                                            "#DISABLED BY UBERENV: " + cfg_scope_stmt)
-        open(spack_lib_config,"w").write(cfg_script)
-
-
-    def patch(self):
-
-        cfg_dir = self.config_dir()
-        spack_dir = self.dest_spack
-
-        # force spack to use only "defaults" config scope
-        self.disable_spack_config_scopes(spack_dir)
-        spack_etc_defaults_dir = pjoin(spack_dir,"etc","spack","defaults")
+    def load(self):
+        # load spack environment, potentially overriding spack defaults
+        # uberenv used to handle defaults overriding, now spack environment can
+        # be used to do so, but it relies on each project to make sure their
+        # environment is well designed.
 
-        # copy in "defaults" config.yaml
-        config_yaml = os.path.abspath(pjoin(self.uberenv_path,"spack_configs","config.yaml"))
-        sexe("cp {} {}/".format(config_yaml, spack_etc_defaults_dir ), echo=True)
+        self.spack_cmd = "{} -e {}".format(self.spack_cmd,self.config_dir())
```

Comment:
What happens if there is no spack config dir? Will this error out?
Perhaps we need to only add the `-e <config-dir>` option when it is available?

Reply:
@kennyweiss an environment is treated the same way config dir were.
It means that Uberenv will try to retrieve the environment corresponding to the current OS/system type.
The behavior when no environment is found is then consistent with what it was before: it throws and error.

This means projects must add a new environment (spack.yaml file) in each config dir.
Adding a basic one is fairly straightforward though, and consists in:
```
spack:
  include:
  - ../config.yaml
  - compilers.yaml
  - packages.yaml

  specs: []

  view: false
  concretization: separately
```
and in compilers.yaml and packages.yaml, changing respectively:
- `compilers:` -> `compilers::`
- `packages:`-> `packages::`

In order to override and scoping (can be done for config too, but I think we will need to discuss the config case separately: overriding scopes in config is not safe).